plugins/notion/src/main.tsx

```diff
@@ -1,94 +1,74 @@
-import "./globals.css"
 import "framer-plugin/framer.css"
 
-import { QueryClient, QueryClientProvider } from "@tanstack/react-query"
 import { framer } from "framer-plugin"
-import React, { type ReactNode, Suspense } from "react"
+import React from "react"
 import ReactDOM from "react-dom/client"
-import { ErrorBoundary } from "react-error-boundary"
-import { App } from "./App"
-import { CenteredSpinner } from "./components/CenteredSpinner"
-import { ErrorBoundaryFallback } from "./components/ErrorBoundaryFallback"
-import { logSyncResult } from "./debug.ts"
-import { getPluginContext, type PluginContext, type PluginContextUpdate, synchronizeDatabase } from "./notion"
-import { assert, syncMethods } from "./utils.ts"
 
-const root = document.getElementById("root")
-if (!root) throw new Error("Root element not found")
+import { App } from "./App.tsx"
+import { PLUGIN_KEYS } from "./api"
+import auth from "./auth"
+import { getExistingCollectionDatabaseIdMap, syncExistingCollection } from "./data"
+import { Authenticate } from "./Login.tsx"
 
-const queryClient = new QueryClient({
-    defaultOptions: {
-        queries: {
-            retry: false,
-        },
-    },
-})
+const activeCollection = await framer.getActiveManagedCollection()
 
-function shouldSyncImmediately(pluginContext: PluginContext): pluginContext is PluginContextUpdate {
-    if (pluginContext.type !== "update") return false
+const tokens = await auth.getTokens()
 
-    if (!pluginContext.database) return false
-    if (!pluginContext.slugFieldId) return false
-    if (pluginContext.hasChangedFields) return false
+const root = document.getElementById("root")
+if (!root) throw new Error("Root element not found")
 
-    return true
+if (!tokens) {
+    await new Promise<void>(resolve => {
```

Comment:
Also non-actionable, comment for the future: This feels off 🤔 Having to mount twice.

Reply:
Can you do this, please? I'm doing it in Airtable.

```ts
if (!tokens) {
    const reactRoot = ReactDOM.createRoot(root)

    await new Promise<void>(resolve => {
        reactRoot.render(
            <React.StrictMode>
                <Authenticate onAuthenticated={resolve} />
            </React.StrictMode>
        )
    })

    reactRoot.unmount()
}
```